src/snowflake/snowpark/_internal/open_telemetry.py

```diff
@@ -19,21 +18,22 @@
 except ImportError:
     open_telemetry_found = False
 
-if open_telemetry_found:
-    tracer = trace.get_tracer("snow.snowpark.dataframe")
-
 
 @contextmanager
 def open_telemetry_context_manager(func, dataframe):
 
     # trace when required package is installed
     if open_telemetry_found:
-        name = func.__qualname__
+        class_name = func.__qualname__
+        name = func.__name__
+        tracer = trace.get_tracer(f"snow.snowpark.{class_name.split('.')[0].lower()}")
         with tracer.start_as_current_span(name) as cur_span:
             try:
```

Comment:
`tracer = trace.get_tracer(f"snow.snowpark.{class_name.split('.')[0].lower()}")` cannot be moved because it needs to be outside with to set tracer, but I do can move other codes into try